src/core/ext/transport/chttp2/client/chttp2_connector.c

```diff
@@ -136,6 +136,8 @@ static void start_handshake_locked(grpc_exec_ctx *exec_ctx,
   c->handshake_mgr = grpc_handshake_manager_create();
   grpc_handshakers_add(exec_ctx, HANDSHAKER_CLIENT, c->args.channel_args,
                        c->handshake_mgr);
+  grpc_endpoint_add_to_pollset_set(exec_ctx, c->endpoint,
```

Comment:
I think we should remove endpoints from the pollset_set after the handshaking process is complete. But we did't do this in other processes (e.g. establishing tcp connection, creating connected subchannels). And we don't even have an interface to do so. I'll check if there is a reason not doing this. If not I'll add a `grpc_endpoint_del_from_pollset_set` interface.